subprojects/docs/src/docs/release/notes.md

```diff
@@ -272,6 +272,33 @@ There are better ways for re-using task logic, for example by using [task depend
 
 The default output location in [EclipseClasspath](dsl/org.gradle.plugins.ide.eclipse.model.EclipseClasspath.html#org.gradle.plugins.ide.eclipse.model.EclipseClasspath:defaultOutputDir) changed from `${project.projectDir}/bin` to `${project.projectDir}/bin/default`.
 
+### Blacklist repository on `IOException`s 
+
+In the event of a timeout (or any `IOException`s which are not caused by error HTTP status code), Gradle will skip subsequent connections to the same repository for the duration of the build. 
+The output of a build clearly indicates which request was skipped.
+
+```
+* What went wrong:
+Could not resolve all files for configuration ':deps'.
+> Could not resolve group:a:1.0.
+  Required by:
+      project :
+   > Could not resolve group:a:1.0.
+      > Could not get resource 'http://localhost:54347/repo/group/a/1.0/a-1.0.pom'.
+         > Could not GET 'http://localhost:54347/repo/group/a/1.0/a-1.0.pom'.
+            > Read timed out
+> Could not resolve group:b:1.0.
+  Required by:
+      project :
+   > Skipped due to earlier error
+```
+
+### Dependency resolution order
+
+Previous versions of Gradle would fall through to the next repository if resolution in one repository failed.
+This behaviour might cause potentially indeterministic resolution result. Now Gradle will explicitly rethrow exceptions which occur in dependency resolution instead of quietly continue to the next repository.
+This may break existed build. In this case, you should fix the error to avoid build indeterminism.
```

Comment:
:x: "This behaviour might cause potentially indeterministic resolution result."

The common word would be "Nondeterministic".

"This may break existed build. In this case, you should fix the error to avoid build indeterminism."

I'd completely remove that sentence. This might just happen due to network issues.

Reply:
Done.